Movecraft/src/main/java/net/countercraft/movecraft/craft/CraftManager.java

```diff
@@ -202,7 +204,27 @@ public void sink(@NotNull Craft craft) {
         if (craft instanceof PlayerCraft)
             playerCrafts.remove(((PlayerCraft) craft).getPilot());
 
-        crafts.add(new SinkingCraftImpl(craft));
+        SinkingCraftImpl sinkingCraft = new SinkingCraftImpl(craft);
+        crafts.add(sinkingCraft);
+
+        int tickDelay = sinkingCraft.getType().getIntProperty(CraftType.EXPLODE_ON_SINK_DELAY);
+        new BukkitRunnable() {
+
+            @Override
+            public void run() {
+                CraftType type = sinkingCraft.getType();
+                float explosionStrength = type.getFloatProperty(CraftType.EXPLODE_ON_SINK);
+                boolean incendiary = type.getBoolProperty(CraftType.INCENDIARY_ON_SINK);
+
+                Location location = sinkingCraft.getHitBox().getMidPoint().toBukkit(sinkingCraft.getWorld());
+
+                ExplosionEvent event = new ExplosionEvent(location, explosionStrength, incendiary);
+                Bukkit.getServer().getPluginManager().callEvent(event);
+
+                if (!event.isCancelled())
+                    location.createExplosion(explosionStrength, incendiary);
+            }
+        }.runTaskLater(Movecraft.getInstance(), tickDelay);
```

Comment:
This implementation does not match the requested feature in #501.  The request was that blocks randomly explode as the craft sinks, this triggers a single explosion at the center.

Reply:
My bad went by assumption, and didnt finish reading, my idea was that if like when a player dies/craft sinks or a craft sinks it explodes so it can't be used by the enemy or for modern servers that use movecraft and are like OilTrucks or stuff that in general can carry dangerous stuff they explode on sink

Reply:
this is too different from the feature in #501, if this features isn't needed I will close it, otherwise I will close the PR